rkyv/src/with/core.rs

```diff
@@ -6,18 +6,183 @@ use crate::{
         ArchivedOptionNonZeroU16, ArchivedOptionNonZeroU32, ArchivedOptionNonZeroU64,
         ArchivedOptionNonZeroU8,
     },
-    with::{ArchiveWith, AsBox, DeserializeWith, Inline, Niche, RefAsBox, SerializeWith, Unsafe},
+    option::ArchivedOption,
+    ser::{ScratchSpace, Serializer},
+    vec::{ArchivedVec, VecResolver},
+    with::{
+        ArchiveWith, AsBox, DeserializeWith, Inline, Map, Niche, RefAsBox, SerializeWith, Unsafe,
+    },
     Archive, ArchiveUnsized, Deserialize, Fallible, Serialize, SerializeUnsized,
 };
 use ::core::{
     cell::{Cell, UnsafeCell},
     convert::TryInto,
+    hint::unreachable_unchecked,
+    marker::PhantomData,
     num::{
         NonZeroI128, NonZeroI16, NonZeroI32, NonZeroI64, NonZeroI8, NonZeroIsize, NonZeroU128,
         NonZeroU16, NonZeroU32, NonZeroU64, NonZeroU8, NonZeroUsize,
     },
+    ptr,
 };
 
+// Map for Vecs
```

Comment:
This part needs to go into `alloc.rs` because `Vec` isn't enabled on `#![no_std]` builds without `extern crate alloc`.

Reply:
fixed